lib/logstash/outputs/sqs.rb

```diff
@@ -1,136 +1,168 @@
+#   * `sqs:GetQueueUrl`
+#   * `sqs:SendMessage`
+#   * `sqs:SendMessageBatch`
 #
-# The "consumer" identity must have the following permissions on the queue:
+# Typically, you should setup an IAM policy, create a user and apply the IAM
+# policy to the user. See http://aws.amazon.com/iam/ for more details on
+# setting up AWS identities. A sample policy is as follows:
 #
-#  * sqs:ChangeMessageVisibility
-#  * sqs:ChangeMessageVisibilityBatch
-#  * sqs:GetQueueAttributes
-#  * sqs:GetQueueUrl
-#  * sqs:ListQueues
-#  * sqs:SendMessage
-#  * sqs:SendMessageBatch
+# [source,json]
+# {
+#   "Version": "2012-10-17",
+#   "Statement": [
+#     {
+#       "Effect": "Allow",
+#       "Action": [
+#         "sqs:GetQueueUrl",
+#         "sqs:SendMessage",
+#         "sqs:SendMessageBatch"
+#       ],
+#       "Resource": "arn:aws:sqs:us-east-1:123456789012:my-sqs-queue"
+#     }
+#   ]
+# }
 #
-# Typically, you should setup an IAM policy, create a user and apply the IAM policy to the user.
-# A sample policy is as follows:
-# [source,ruby]
-#      {
-#        "Statement": [
-#          {
-#            "Sid": "Stmt1347986764948",
-#            "Action": [
-#              "sqs:ChangeMessageVisibility",
-#              "sqs:ChangeMessageVisibilityBatch",
-#              "sqs:GetQueueAttributes",
-#              "sqs:GetQueueUrl",
-#              "sqs:ListQueues",
-#              "sqs:SendMessage",
-#              "sqs:SendMessageBatch"
-#            ],
-#            "Effect": "Allow",
-#            "Resource": [
-#              "arn:aws:sqs:us-east-1:200850199751:Logstash"
-#            ]
-#          }
-#        ]
-#      }
+# = Batch Mode
+# When publishing messages to SQS in batches, the following restrictions must
+# be considered:
 #
-# See http://aws.amazon.com/iam/ for more details on setting up AWS identities.
+#   - The maximum allowed individual message size is 256KiB.
+#   - The maximum total payload size (i.e. the sum of all a batch's individual
+#     message lengths) is also 256KiB.
+#
+# When operating in batch mode, this plugin will dynamically adjust the size of
+# the batch to ensure that the total payload size does not exceed 256KiB. If,
+# however, any single message exceeds the 256KiB then it will be dropped.
 #
 class LogStash::Outputs::SQS < LogStash::Outputs::Base
-  include LogStash::PluginMixins::AwsConfig
-  include Stud::Buffer
+  include LogStash::PluginMixins::AwsConfig::V2
 
-  config_name "sqs"
+  config_name 'sqs'
+  default :codec, 'json'
 
-  # Name of SQS queue to push messages into. Note that this is just the name of the queue, not the URL or ARN.
-  config :queue, :validate => :string, :required => true
+  concurrency :shared
 
-  # Set to true if you want send messages to SQS in batches with `batch_send`
-  # from the amazon sdk
+  # Set to `true` to send messages to SQS in batches. The size of the batch is
+  # configurable via `batch_events`.
   config :batch, :validate => :boolean, :default => true
```

Comment:
OK. Should `batch_timeout` be marked as `:obsolete` rather than `:deprecated`? I couldn't really tell what the difference was from looking at the code.